test/multi_index_search_test.dart

```diff
@@ -25,13 +25,13 @@ void main() {
 
     test("Multi search from 2 indexes", () async {
       final result = await client.multiSearch(MultiSearchQuery(queries: [
-        SearchQuery(
+        IndexSearchQuery(

```

Comment:
I think my last point is about having a different class to represent the object that will be created to be sent in the `multiSearch`. I would like to have the same `SearchQuery` for both, this way, we can keep some level of convenience. 

Sometimes I struggle with that type of decision because I completely understand the mindset behind the decision, but in fact, when we create tools that are going to be used for other developers, I try to address the challenges with one thing in mind: "how can I make this thing in the simplest as possible?"

Bringing to the current situation, people already have the overhead to remember the `MultiSearchQuery`, and they will also have to remember the `IndexSearchQuery`. I don't think it matches that mindset (I mentioned earlier).

To be honest, if we could have just the method `multiSearch([SearchQuery(), SearchQuery()])` would be the best in my POV 😄, WDYT?

Reply:
the problem with that is people can call `SearchQuery` without passing `indexUid`, which will give them a runtime error.
by creating a more strict class, we ensure static typing for the indexUid parameter, meaning that the user can never pass it as null.

Personally, I would rather remember an extra class than to forget `indexUid` and not know about it until later.

Reply:
actually with the current state of the PR, we can move the `q` parameter from the `SearchQuery` to the `IndexSearchQuery`

Reply:
yeah, the static errors will play a huge part in this approach.